pkg/signature/kms/cliplugin/README.md

```diff
@@ -0,0 +1,83 @@
+# CLI Plugin
+
+This is a package and module for using custom KMS plugins as separate executables. 
+It is intended to be used by cosign, but you may use this in your own programs that import sigstore.
+
+## Design
+
+We follow [kubectl's style](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/#writing-kubectl-plugins) of plugins. Any language that can create CLI programs can be used to make a plugin.
+
+### Usage
+
+Plugins are separate programs on your system's PATH, named in the scheme `sigstore-kms-[name]`, like `sigstore-kms-my-hsm`. They can be invoked with cosign like `cosign [sub-command] --key "my-hsm://my-key-id" ...
```

Comment:
For now, if there is a clash, it will run the one that appears first in PATH. It won't error due to a clash. We could allow the user to specify the location of the executable with an env variable.